src/lib/UI/Module/ContentTree/NodeFactory.php

```diff
@@ -255,4 +222,89 @@ private function getSortClauses(
             return []; // rely on storage engine default sorting
         }
     }
+
+    /**
+     * @param \eZ\Publish\API\Repository\Values\Content\ContentInfo[] $contentInfoList
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     */
+    private function buildNode(
+        Location $location,
+        ArrayObject $contentInfoList,
+        ?LoadSubtreeRequestNode $loadSubtreeRequestNode = null,
+        bool $loadChildren = false,
+        int $depth = 0,
+        ?string $sortClause = null,
+        string $sortOrder = Query::SORT_ASC
+    ): Node {
+        $contentInfo = $location->getContentInfo();
+        $contentId = $location->contentId;
+        if (!isset($this->contentInfoList[$contentId])) {
+            $contentInfoList[$contentId] = $contentInfo;
+        }
+
+        // Top Level Location (id = 1) does not have a Content Type
+        $contentType = $location->depth > 0
+            ? $contentInfo->getContentType()
+            : null;
+
+        $limit = $this->resolveLoadLimit($loadSubtreeRequestNode);
+        $offset = null !== $loadSubtreeRequestNode
+            ? $loadSubtreeRequestNode->offset
+            : 0;
+
+        $children = [];
+        if ($depth < $this->getSetting('tree_max_depth') && $loadChildren) {
```

Comment:
Checking against boolean value goes first, as it's a simpler and faster operation.

```suggestion
        if ($loadChildren && $depth < $this->getSetting('tree_max_depth')) {
```

It's a copy from original though I see, so you can leave as is if you prefer.